Automattic/wp-calypso
# Unify code seam for the Global View sidebar

Related to https://github.com/Automattic/wp-calypso/pull/87583

## Proposed Changes

This PR modifies the approach to retrieve the flag for the Global View sidebar by utilizing a selector due to the following reasons:

- A hook [cannot be used](https://react.dev/warnings/invalid-hook-call-warning) in `connect()` function. The previous usage of `useGlobalSidebar` was feasible because it was not genuinely a hook but merely a function with a `use**` prefix. This could lead to misunderstandings, as it suggests hook-like behavior, which is not supported in this context (e.g., utilizing `useState` would result in error).
- The flag for the Global Site View has transitioned to using a selector, as in https://github.com/Automattic/wp-calypso/pull/87583.
- We want to have less code seam to avoid overheads: https://github.com/Automattic/dotcom-forge/issues/5609

## Testing Instructions



* Prepare a site with the classic view
* Open Calypso Live
* Go to /sites
* Go to /home/<your-site>
* Go to /me
* Go to /read
* Ensure the masterbar is hidden and the Global Nav is shown on all the pages above

## Pre-merge Checklist
